pkg/operator/operator.go

```diff
@@ -591,12 +607,7 @@ func (o *Operator) sync(ctx context.Context, key string) error {
 	config.SetTelemetryMatches(o.telemetryMatches)
 	config.SetRemoteWrite(o.remoteWrite)
 
-	var proxyConfig manifests.ProxyReader
-	proxyConfig, err = o.loadProxyConfig(ctx)
-	if err != nil {
-		klog.Warningf("using proxy config from CMO configmap: %v", err)
-		proxyConfig = config
-	}
+	var proxyConfig = getProxyReader(func() (*ProxyConfig, error) { return o.loadProxyConfig(ctx) }, config)
```

Comment:
rather than a new function we could extend `loadProxyConfig()` to read proxy configuration from the configmap (if it exists) and fallback to the cluster-wide proxy settings?

```
proxyConfig, err = o.loadProxyConfig(ctx, config)
```

Reply:
A downside of that is that then `loadProxyConfig` would have 2 concerns: fetching the cluster wide config, which is already somewhat complex due to state handling for `o.lastKnowProxyConfig`; and deciding the precedence between those two configurations. 
Also `loadProxyConfig` is difficult to test because the version of the openshift client in `vendor` doesn't have support for faking for the config clientset:

```

$ find vendor/github.com/openshift/client-go -type d -wholename '*/versioned/fake'
vendor/github.com/openshift/client-go/route/clientset/versioned/fake
vendor/github.com/openshift/client-go/security/clientset/versioned/fake
$
```

which explains why there is no unit test for `loadProxyConfig`. Even though this is a small issue, it has significant impact in the functionality, so it would be nice to have it covered by a unit test to prevent (or at least difficult) future regressions.

I was thinking on renaming `loadProxyConfig` to `loadClusterProxyConfig` to make all more explicit. What do you think? 

In any case I'm not adamant with this, and I'm ok with moving the new logic to `loadProxyConfig` if you think that would make the code simpler, and it is ok not adding a unit test for this

Reply:
> Also loadProxyConfig is difficult to test because the version of the openshift client in vendor doesn't have support for faking for the config clientset.

You should be able to import "github.com/openshift/client-go/config/clientset/versioned/fake" from the test file then run `go mod vendor` to update the vendor/ directory.

Totally agree that we need unit tests to avoid regressoins. I'm also not stuck on my recommendation so do as you feel :)